### ISSUE TITLE
Move networking classes from panda3d.core into separate panda3d.net

### ISSUE THREAD

**Issue by rdb** [panda3d/panda3d#1466]
It's possible to build Panda without networking code.  However, this causes a bunch of classes to go missing from panda3d.core, causing various imports within direct to stop working.  It's not at all obvious to figure out which code depends on net classes and which code doesn't.

I think the networking code should be in a separate panda3d.net module so that it's obvious when something relies on net code, and results in less cryptic import errors.  This additional modularity would mean that it can be trivially detected whether net code is used and it can be easily removed at deployment if it is not used.

This would unfortunately be a breaking change.  But it is a fairly easy one to work around.

I would suggest 1.11 to be a transition release where the classes are defined in both panda3d.net and panda3d.core, and removing them from panda3d.core in the release after that.

For what it's worth, I ran into this when building for web, where the networking code is compiled out entirely (since raw sockets aren't available on the web).  In other situations we tend to create stub versions of classes that do nothing but this would be impractical for the networking classes.

**Comment by BMaxV** [comment]
Good idea, I don't think the code/docs are usable anyway. Is it known whether there are projects that would be affected?
